sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileSourceStrategy.scala

```diff
@@ -97,7 +98,16 @@ object FileSourceStrategy extends Strategy with Logging {
         dataColumns
           .filter(requiredAttributes.contains)
           .filterNot(partitionColumns.contains)
-      val outputSchema = readDataColumns.toStructType
+      val outputSchema = if (fsRelation.sqlContext.conf.isParquetNestColumnPruning) {
```

Comment:
All three tests failed because a datetime check error, correct answer is like '2015-05-23' but the spark answer is '2015-05-22', I don't think this error is made by my patch.
Do anybody have the same problem before? it really confusing me and running test suit seperate it will pass!